src/Element/Select2.php

```diff
@@ -46,6 +48,25 @@ public static function processSelect(&$element, FormStateInterface $form_state,
       $element['#options'] = $empty_option + $element['#options'];
     }
 
+    if ($element['#autocreate'] && $element['#target_type']) {
+      $options = $element['#selection_settings'] + [
+        'target_type' => $element['#target_type'],
+        'handler' => $element['#selection_handler'],
+      ];
+      /** @var \Drupal\Core\Entity\EntityReferenceSelection\SelectionInterface $handler */
+      $handler = \Drupal::service('plugin.manager.entity_reference_selection')->getInstance($options);
+      foreach ($element['#value'] as $id) {
+        if (!isset($element['#options'][$id]) && $handler instanceof SelectionWithAutocreateInterface) {
```

Comment:
$handler instanceof SelectionWithAutocreateInterface should be tested once outside of the loop